keps/993-two-phase-admission/README.md

```diff
@@ -253,6 +254,27 @@ The controller implementing a particular check should:
 * After approving the workload, keep an eye on the check if it starts failing,
   fail the check and cause the workload to move back to the suspended state.
 
+### Preemption Admission Check Controller
+
+Because now, the time when the eviction of the preemption candidate should vary based on the preemptor state
+the scheduler will not issue the eviction during it's process instead it will set a `Pending` admission check
+that is manged by a new built-in admission check controller.
+
+The **Admission Check Controller** will:
+
+- Watch for a change in state of the workloads pending preemption.
+- Watch for workloads that are finishing execution and therefore releasing quota.
+- Watch for AdmissionCheck changes, since the preemption policy can change.
+
+Since evaluating the current preemption state and the needs of a workload requires checking the sate of all resource pool the events are rate limited the kueue's cache is used.
+
+At every run the controller will get the list of workloads pending preemption.
+Since for some of these workloads is not necessary to issue eviction at that given point (eg. Having a pending check that uses AfterCheckPassedOrOnDemand policy) their quota reservation will be ignored.
```

Comment:
so we will get a snapshot that doesn't contain the workloads with pending preemptions, right?

Reply:
The workloads that don't require to issue evictions at that particular point in time are removed from the snapshot.

Reply:
Oh I see. So if there is a workload that requires preemptions and it's ready for preemption, it itself could be preempted (it's QuotaReserved would be cleared). Am I understanding correctly?

An alternative implementation (maybe weaker) would be that we only take into consideration in the snapshot the workloads that are fully admitted (not just quota reserved).